app/controllers/uploads_controller.rb

```diff
@@ -22,16 +22,8 @@ def create
 
     begin
       @upload.check_for_headers
-      failed = load_csv.failed_instances
-
-      validation_warnings = failed.sort { |a, b| a.errors[:row].first.to_i <=> b.errors[:row].first.to_i }
-                                  .map(&:display_errors_with_row)
-      header_warnings = @upload.all_warnings
-
-      flash.alert = { 'The upload succeeded: ' => @upload.csv_type }
-
-      flash.alert['The following rows should be checked: '] = validation_warnings unless validation_warnings.empty?
-      flash.alert['The following headers should be checked: '] = header_warnings unless header_warnings.empty?
```

Comment:
My preference is to see these flash message assignments directly in the controller action. The value of those messages can be offloaded into some other method but I think it's valuable to immediately see that this response includes flash messages.